devel/bitwise/Portfile

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        mellowcandle bitwise v0.30
```

Comment:
We don't want the "v" in the version number. So change this line to:

`github.setup        mellowcandle bitwise 0.30 v
`

Reply:
I initially did that, as it's in the docs. However, the URL is not constructed properly this way and downloading the source fails:

```
--->  Fetching distfiles for bitwise
--->  bitwise-0.30.tar.gz does not exist in /opt/local/var/macports/distfiles/bitwise
--->  Attempting to fetch bitwise-0.30.tar.gz from https://github.com/mellowcandle/bitwise/releases/download/v0.30
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from https://distfiles.macports.org/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://nue.de.distfiles.macports.org/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://lil.fr.distfiles.macports.org/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://mse.uk.distfiles.macports.org/sites/distfiles.macports.org/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   269  100   269    0     0   5322      0 --:--:-- --:--:-- --:--:--  5380
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://fco.it.distfiles.macports.org/mirrors/macports-distfiles/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://ykf.ca.distfiles.macports.org/MacPorts/mpdistfiles/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://ywg.ca.distfiles.macports.org/mirror/macports/distfiles/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://aus.us.distfiles.macports.org/macports/distfiles/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://jog.id.distfiles.macports.org/macports/distfiles/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://jnb.za.distfiles.macports.org/distfiles/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://kmq.jp.distfiles.macports.org/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://cjj.kr.distfiles.macports.org/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://aarnet.au.distfiles.macports.org/pub/macports/distfiles/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from https://pek.cn.distfiles.macports.org/macports/distfiles/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
--->  Attempting to fetch bitwise-0.30.tar.gz from http://nou.nc.distfiles.macports.org/pub/macports/distfiles.macports.org/bitwise
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
Error: Failed to fetch bitwise: The requested URL returned error: 404 Not Found
Error: See /opt/local/var/macports/logs/_Users_mbischoff_ports_devel_bitwise/bitwise/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets to report a bug.
Error: Processing of port bitwise failed
```

Reply:
You also set `github.tarball_from releases` and that is not normally needed. Remove that line and try again.

Reply:
This was done because the release tarballs can just run the usual configure/make/make install cycle whereas the tagged version needs to run a bootstrap script (contains an autoreconf command).

Reply:
Hmm. What is the actual download URL for the "releases" tarball?

Compare it to the attempted one of:
`https://github.com/mellowcandle/bitwise/releases/download/v0.30/bitwise-0.30.tar.gz
`

Reply:
It's `https://github.com/mellowcandle/bitwise/releases/download/v0.30/bitwise-v0.30.tar.gz`

MacPorts is missing the `v` in the filename. It's probably because the file is manually named and not simply generated by GitHub.

What do you think is the best approach here?

Reply:
That appears to be a bug in the GitHub portgroup. I'm going to see what other ports do in the same situation.

Reply:
Just found the `use_autoreconf` in the docs. I'm just going use it to effectively run the bootstrap command. Problem solved. ;)

Reply:
One fix would be to set the distname manually:

`distname ${name}-v${version}
`

Reply:
.... You said it is manually named so we have to set the `distname` manually like this.

Reply:
Alright. I'm up for whatever works and is in line with the spirit of MacPorts. Just let me know which one you'd prefer: setting a manual distname or use_autoreconf keyword? I tested both and both work.